qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -649,17 +649,26 @@ def on_current_changed(self, idx):
             log.webview.debug("on_current_changed got called with invalid "
                               "index {}".format(idx))
             return
+        if self._now_focused and config.val.tabs.mode_on_change == 'restore':
+            current_mode = modeman.instance(self._win_id).mode
```

Comment:
I wonder if there's a better way to get the current mode, this seems fine to me, but I just wanted to flag it

Reply:
That's how it's done elsewhere in the code, don't know if there is another method...

Reply:
Seems fine to me.